lib/ui/button.tsx

```diff
@@ -30,19 +32,24 @@ const buttonVariants = cva(
         promote:
           "border border-solid border-f1-border-promote bg-f1-background-promote text-f1-foreground hover:bg-f1-background-promote-hover",
       } satisfies Record<(typeof variants)[number], string>,
+      size: {
+        small: "h-6 rounded-sm px-2",
+        medium: "h-8 rounded-md px-3",
+        large: "h-10 rounded-md px-4",
+      } satisfies Record<(typeof sizes)[number], string>,
       round: {
         true: "aspect-square px-0",
-        false: "px-4",
       },
     },
     defaultVariants: {
       variant: "default",
+      size: "medium",
       round: false,
     },
   }
 )
 
-const iconVariants = cva("transition-colors", {
+const iconVariants = cva("-ml-0.5 transition-colors", {
```

Comment:
This is a visual adjustment for the icon. When the icon is present, the left padding of the button should be a bit lower. Not sure if this is the best approach though.

Reply:
This is weird, no? Maybe it's because of the box model? Let's merge this and I can check out what's happening :)

Reply:
It's weird indeed, haha. The basic idea is: the button has a left padding of 8px (for example), but when the icon is present, it should be 6px. This is a hacky way of getting that result but is, well, hacky 😅

Reply:
Aaaaaah I understand. I think we can conditionally change this on the button itself when we do have an `icon` property instead of doing it on the icon. What do you think?

Reply:
The component that knows if there’s an icon is our Button, but the styling is here in the shadcn button, so I wasn’t exactly sure how to handle that. 😬

Reply:
Aaaaah dammit. Will give it a look.